flycheck.el

```diff
@@ -4257,6 +4263,27 @@ universal prefix arg, and only the id with normal prefix arg."
     (`(eval ,_) t)
     (_ nil)))
 
+(defun flycheck-default-directory-wrapper (checker)
+  "Function returning apropriate current working directory.
+
+Work out the working directory from which the CHECKER command
+will be called.  The default is the `default-directory' of the source-buffer
+from which flycheck is calling the checker.
+The :default-directory must be a function that will return an absolute path."
+  (let* ((cwd (flycheck-checker-get checker 'default-directory))
+         (default-directory (if cwd
+                                (if (functionp cwd)
+                                    (funcall cwd)
+                                  (error "Syntax Checker %S has defined :default-directory which is not a function: %s"
```

Comment:
Even better shorten the error message ":default-directory %S of syntax checker %S is not a function".